engine/src/vega_cast_utils.h

```diff
@@ -91,20 +91,17 @@ inline long locale_aware_stol(const std::string& s, const long default_value = 0
 }
 
 template<class TargetType, class SourceType>
-inline TargetType* vega_dynamic_cast_ptr(SourceType* from) {
+inline TargetType* vega_dynamic_cast_ptr(SourceType* from, const bool allow_null = false) {
```

Comment:
@royfalk This parameter, and [the commit](https://github.com/vegastrike/Vega-Strike-Engine-Source/pull/1416/commits/392589c20aeb3ffb770488522bcf0ffc145db65f) in which it was introduced, are new in this PR as opposed to the non-SDL3 version. (I plan to port this over there too.)